catalog/core/catalog-core-standardframework/src/test/java/org/codice/ddf/catalog/resource/download/impl/ResourceDownloadTest.java

```diff
@@ -0,0 +1,261 @@
+        setupMockResourceCacheServiceMBean(false);
+        setupMockDownloadManager(false);
+        ResourceDownloadMBean resourceDownloadMBean = createResourceDownloadMBean();
+
+        // Perform Test
+        resourceDownloadMBean.downloadToCache(SOURCE_ID, METACARD_ID);
+    }
+
+    @Test(expected = MBeanException.class)
+    public void downloadToCacheOnlyWhenCatalogFrameworkThrowsIOException() throws Exception {
+        // Setup
+        setupMockResourceCacheServiceMBean(true);
+        setupMockDownloadManager(true);
+        setupMockCatalogFramework(mockResourceResponse, IOException.class);
+        ResourceDownloadMBean resourceDownloadMBean = createResourceDownloadMBean();
+
+        // Perform Test
+        resourceDownloadMBean.downloadToCache(SOURCE_ID, METACARD_ID);
+    }
+
+    @Test(expected = MBeanException.class)
+    public void downloadToCacheOnlyWhenCatalogFrameworkThrowsResourceNotSupportedException()
+        throws Exception {
+        // Setup
+        setupMockResourceCacheServiceMBean(true);
+        setupMockDownloadManager(true);
+        setupMockCatalogFramework(mockResourceResponse, ResourceNotSupportedException.class);
+        ResourceDownloadMBean resourceDownloadMBean = createResourceDownloadMBean();
+
+        // Perform Test
+        resourceDownloadMBean.downloadToCache(SOURCE_ID, METACARD_ID);
+    }
+
+    @Test(expected = MBeanException.class)
+    public void downloadToCacheOnlyWhenCatalogFrameworkThrowsResourceNotFoundException()
+        throws Exception {
+        // Setup
+        setupMockResourceCacheServiceMBean(true);
+        setupMockDownloadManager(true);
+        setupMockCatalogFramework(mockResourceResponse, ResourceNotFoundException.class);
+        ResourceDownloadMBean resourceDownloadMBean = createResourceDownloadMBean();
+
+        // Perform Test
+        resourceDownloadMBean.downloadToCache(SOURCE_ID, METACARD_ID);
+    }
+
+    @Test
+    public void downloadToCacheOnly() throws Exception {
+        // Setup
+        setupMockResourceCacheServiceMBean(true);
+        setupMockDownloadManager(true);
+        setupMockCatalogFramework(mockResourceResponse, null);
+        ResourceDownloadMBean resourceDownloadMBean = createResourceDownloadMBean();
+
+        // Perform Test
+        String response = resourceDownloadMBean.downloadToCache(SOURCE_ID, METACARD_ID);
+
+        assertThat(response, is(String.format(SUCCESS_MESSAGE_TEMPLATE, METACARD_ID, SOURCE_ID)));
+    }
+
+    @Test
+    public void testInit() throws Exception {
+        // Setup
+        setupMockMBeanServer(false, false);
+        ResourceDownload resourceDownloadMBean = (ResourceDownload) createResourceDownloadMBean();
+
+        // Perform Test
+        resourceDownloadMBean.init();
+
+        verify(mockMBeanServer).registerMBean(any(StandardMBean.class),
+                eq(resourceDownloadObjectName));
+    }
+
+    @Test
+    public void testInitMBeanAlreadyRegistered() throws Exception {
```

Comment:
We _really_ need to create a common class for all that MBean boiler plate code and get rid of all those repeated tests...